org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/utils/DurationUtils.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2024 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.core.internal.utils;
+
+import java.time.Duration;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class DurationUtils {
+  private DurationUtils() {
+    // utility class
+  }
+
+  /**
+   *  Users can provide properties containing simple timeouts ("3" is "3 minutes") or more complex ones ("PT4H" is
+   *  "4 hours"). We have to check all the different cases for users who configure the timeouts differently.
+   */
+  @Nullable
+  public static Duration getTimeoutProperty(String propertyName) {
+    var property = System.getProperty(propertyName);
+    if (property == null) {
+      return null;
+    }
+
+    try {
+      return Duration.ofMinutes(Integer.parseInt(property));
+    } catch (NumberFormatException ignored) {
+      return Duration.parse(property);
```

Comment:
There is a `DateTimeParseException` thrown in case it's wrong, shouldn't it be safer to catch it and return `null`?

Reply:
I re-used the old logic that got lost (see [ticket comment](https://sonarsource.atlassian.net/browse/SLE-851?focusedCommentId=604251)) from SLCORE.

I think you're right. I will change it that it is logged and then returns `null`. I wouldn't fail hard in this case but for any means necessary the user should be informed the input was garbage :smile: